app/model/SubscriptionAcquisitionComponents.scala

```diff
@@ -0,0 +1,181 @@
+          case JsSuccess(referrerAcquisitionData, _) => Some(referrerAcquisitionData)
+          case e: JsError => {
+            logger.warn(s"Unable to decode JSON $parsedJson to an instance of ReferrerAcquisitionData. ${JsError.toJson(e).toString()}")
+            None
+          }
+        })
+    }
+
+    private def printOptionsFromPaperData(p: PaperData): Option[PrintOptions] = {
+      // Explicit imports because Product and Benefit have name collisions
+      import com.gu.memsub.Product.{Delivery, Voucher, Weekly}
+      import com.gu.memsub.Benefit.{MondayPaper, TuesdayPaper, WednesdayPaper, ThursdayPaper, FridayPaper, SaturdayPaper, SundayPaper, Digipack}
+
+      val paperDays = p.plan.charges.benefits.list.collect {
+        case p: PaperDay => p
+      }
+
+      sealed trait When
+      case object Sunday extends When
+      case object Weekend extends When
+      case object Sixday extends When
+      case object Everyday extends When
+
+      val when = Some(paperDays.sortBy(_.dayOfTheWeekIndex)) collect {
+        case List(SundayPaper) => Sunday
+        case List(SaturdayPaper, SundayPaper) => Weekend
+        case List(MondayPaper, TuesdayPaper, WednesdayPaper, ThursdayPaper, FridayPaper, SaturdayPaper) => Sixday
+        case List(MondayPaper, TuesdayPaper, WednesdayPaper, ThursdayPaper, FridayPaper, SaturdayPaper, SundayPaper) => Everyday
+      }
+
+      val hasDigipack = p.plan.charges.benefits.list.contains(Digipack)
+
+      val printProduct = Some(when, p.plan.product, hasDigipack) collect {
+        case (Some(Sunday), Delivery, false) => PrintProduct.HomeDeliverySunday
+        case (Some(Sunday), Delivery, true) => PrintProduct.HomeDeliverySundayPlus
+        case (Some(Weekend), Delivery, false) => PrintProduct.HomeDeliveryWeekend
+        case (Some(Weekend), Delivery, true) => PrintProduct.HomeDeliveryWeekendPlus
+        case (Some(Sixday), Delivery, false) => PrintProduct.HomeDeliverySixday
+        case (Some(Sixday), Delivery, true) => PrintProduct.HomeDeliverySixdayPlus
+        case (Some(Everyday), Delivery, false) => PrintProduct.HomeDeliveryEveryday
+        case (Some(Everyday), Delivery, true) => PrintProduct.HomeDeliveryEverydayPlus
+
+        case (Some(Sunday), Voucher, false) => PrintProduct.VoucherSunday
+        case (Some(Sunday), Voucher, true) => PrintProduct.VoucherSundayPlus
+        case (Some(Weekend), Voucher, false) => PrintProduct.VoucerWeekend
+        case (Some(Weekend), Voucher, true) => PrintProduct.VoucerWeekendPlus
+        case (Some(Sixday), Voucher, false) => PrintProduct.VoucherSixday
+        case (Some(Sixday), Voucher, true) => PrintProduct.VoucherSixdayPlus
+        case (Some(Everyday), Voucher, false) => PrintProduct.VoucherEveryday
+        case (Some(Everyday), Voucher, true) => PrintProduct.VoucherEverydayPlus
+
+        case (_, _: Weekly, false) => PrintProduct.GuardianWeekly
+        case (_, _: Weekly, true) => PrintProduct.GuardianWeeklyPlus
```

Comment:
the thing that scares me about this is this information should be in the Catalog, but it's been explicitly not represented there (for better or worse) and now it's hard coded somewhere non-resuable which may not be updated if the actual catalog changes.
ideally the catalog would be updated to actually know about all the different products and you could just read it from there, but clearly that's a more scary refactor.

Also, does guardian weekly plus even exist?

Reply:
It's a very good point. But most changes to the catalog will also require us to update our Thrift definitions anyway, so even if the catalog code could automatically handle a change in the mapping of delivery days to print products, our tracking would still probably break if we didn't know about these changes. I think a better approach is to ensure that the Acquisitions team know about any significant changes to the subscription products on offer

Reply:
Also, I've removed Guardian Weekly Plus - you're right, it doesn't exist!